### PR TITLE
fix: fqdn in name is ignored and mrack guesses the name instead #237

### PR DESCRIPTION
**Pull request by jakub-vavra-cz**

### CODE REVIEW

File: tests/unit/test_utils.py [hunk 1]
```diff
@@ -16,6 +16,17 @@ class TestPytestMrackUtils:
     def test_get_shortname(self, hostname, expected):
         assert get_shortname(hostname) == expected
 
+    @pytest.mark.parametrize(
+        "hostname,domain,expected",
+        [
+            ("abc.test", "bar", "abc.test"),
```
Comment (pvoborni) [inline tests/unit/test_utils.py:22]: A host under domain `bar` cannot have a FQDN not containing the domain `bar`. In this case the the `abc.test` must be either under TLD domain `test` or the FQDN should be e.g. `abc.test.bar`

File: src/mrack/outputs/ansible_inventory.py [hunk 2]
```diff
@@ -143,7 +144,7 @@ def create_ansible_host(self, name):
             "ansible_host": ansible_host,
             "ansible_python_interpreter": python,
             "ansible_user": ansible_user,
-            "meta_fqdn": f"{get_shortname(name)}.{dom_name}",
+            "meta_fqdn": f"{get_fqdn(name, dom_name)}",
```
Comment (pvoborni) [inline src/mrack/outputs/ansible_inventory.py:147]: It must be ensured that `meta_fqdn` = `meta_hostname` + `meta_domain`.

**Review by pvoborni — CHANGES_REQUESTED** [review]
Hi, this patch is breaking some behaviors expected from mrack.  

I think it would be good to agree first on what is the expected behavior - e.g. in the issue description.

File: tests/unit/test_utils.py (hunk 1 above)
Comment (jakub-vavra-cz) [reply tests/unit/test_utils.py:22]: Point of this test is that when name is fqdn, domain is ignored/overridden by whatever is in the hostname.
**bar** is there only to make it clear.

File: src/mrack/outputs/ansible_inventory.py (hunk 2 above)
Comment (jakub-vavra-cz) [reply src/mrack/outputs/ansible_inventory.py:147]: The intent here is to be able to override this assumption and have hosts from subdomains directly in the same tld domain by specifying names as fqdn.
```
domains:
  - name: test
    type: sssd
    hosts:
      - name: client.test
        group: medium
        role: client
        os: rhel-9.2
      - name: dns.test
        group: xsmall
        role: dns
        os: rhel-9.2
      - name: master.ldap.test
        group: medium
        role: ldap
        os: rhel-9.2
      - name: master.ipa.test
        group: medium
        role: ipa
        os: rhel-9.2
```

Comment (Tiboris) [reply src/mrack/outputs/ansible_inventory.py:147]: i would expect domain name to be: something.com

Comment (pvoborni) [reply src/mrack/outputs/ansible_inventory.py:147]: To keep discussion on one place, let's continue the discussion in the issue: https://github.com/neoave/mrack/issues/237

Comment (jakub-vavra-cz) [reply src/mrack/outputs/ansible_inventory.py:147]: Resolved.

File: tests/unit/test_utils.py (hunk 1 above)
Comment (jakub-vavra-cz) [reply tests/unit/test_utils.py]: Resolved.

**Review by pvoborni — APPROVED** [review]
This behavior looks good to me. Tests looks good as well. Thanks.

 Maybe @Tiboris  can look at it as well before we merge.

There might also some further improvements done, but they don't block the MR (can be done separately):

- fail early with the error, i.e. before we start provision and not at the end. This could be achieved by e.g. extending https://github.com/neoave/mrack/blob/main/src/mrack/actions/up.py#L64 
- the method passes even if host's name == domain name :)  - it's quite a corner case so it doesn't worry me much

**Review by Tiboris — APPROVED** [review]
Thanks for the patch

**Pull request merged**